app/assets/javascripts/player_listeners.js

```diff
@@ -536,3 +538,30 @@ function handleCreateTimelineModalShow() {
     $('#new-timeline-form')[0].submit();
   });
 }
+
+/**
+ * Handler to refresh stream tokens via reloading the m3u8 file
+ */
+function initM3U8Reload(player, mediaObjectId, sectionIds, sectionShareInfos) {
+  if (player && player.player != undefined) {
+    if (firstLoad === true) {
+      player.player.on('pause', () => {
+        currentTime = player.player.currentTime();
+        // How long to wait before resetting stream tokens: default 5 minutes
+        intervalLength = 5*60*1000
+        reloadInterval = setInterval(m3u8Reload, intervalLength);
+      });
+
+      player.player.on('play', () => {
+        if (reloadInterval !== false) {
+          clearInterval(reloadInterval);
+          reloadInterval = false;
+        }
+      });
+
+      player.player.on('waiting', () => {
+        m3u8Reload();
```

Comment:
Does this work with just fetch or does the player need the sources set again in this case?

Reply:
It was working with just the fetch when I was messing with things this morning.